### DISCUSSION TITLE
Cannot read properties of undefined (reading 'replace')

### DISCUSSION THREAD

**Discussion by drydenwilliams** [system-ui/theme-ui#2014] [Q&A] [answered]
I just can't seem to get `theme-ui` and `Prism` working together at all, I'm just stuck on this error:

When I have: 

```
  pre: ({ children }) => <>{children}</>,
  code: Prism,
```

All I see is this error:

<img width="1440" alt="Screenshot 2021-11-19 at 17 51 32" src="https://user-images.githubusercontent.com/4403089/142668827-4ed4d9e2-cbc4-4d85-abab-8dc1371f86c5.png">

I'm using it with `gatsby` and my versions are: 

```
{
    "@theme-ui/color": "^0.11.3",
    "@theme-ui/prism": "^0.11.3",
    "gatsby-plugin-theme-ui": "^0.11.3",
    "theme-ui": "^0.11.3",
}
```

I've tried updating packages with no luck. And I copied this [setup](https://github.com/PaulieScanlon/gatsby-theme-terminal/blob/main/%40pauliescanlon/gatsby-theme-terminal/package.json) as it's working just fine. 

Any help would be awesome. Thanks :)

**Comment by drydenwilliams** [comment]
I think It might be because it doesn't have a `language` that is being passed through? Or that it should be a JSX element?

```
  code: (props) => {
    console.log("PROPS", props);
    return <Prism {...props} />;
  },
  
// => { children: "color: #444\n" } 
```

**Comment by hasparus** [comment, marked as answer]
`<ThemeUIPrism />` doesn't take `language` prop, it grabs the language from the className, so when the className is missing this can fail.

```
const [language] = outerClassName.replace(/language-/, '').split(' ')
```

I think we should fix it and default missing `className` to an empty string here.

**Reply by hasparus** [reply]
I created an issue to track this bug: https://github.com/system-ui/theme-ui/issues/2016.

You can work around it by setting the default className to `""`.
